src/folderlistmodel.cpp

```diff
@@ -333,6 +339,65 @@ int FolderListModel::standardFolderIndex(FolderStandardType folderType)
     return -1;
 }
 
+bool FolderListModel::lessThan(const QMailFolderId &idA, const QMailFolderId &idB)
+{
+    Q_ASSERT(idA.isValid());
+    Q_ASSERT(idB.isValid());
+
+    QMailFolder aFolder(idA), bFolder(idB);
+    if (aFolder.parentFolderId() == bFolder.parentFolderId()) {
+        // Siblings
+        return aFolder.displayName().compare(bFolder.displayName(), Qt::CaseInsensitive) < 0;
+    } else if (aFolder.parentAccountId() != bFolder.parentAccountId()) {
+        // Different accounts, we still want to compare since local storage
+        // can contain some of the standard folders for the account
+        qCWarning(lcGeneral) << Q_FUNC_INFO << "Comparing folders from different accounts";
+        return aFolder.displayName().compare(bFolder.displayName(), Qt::CaseInsensitive) < 0;
```

Comment:
I'm not sure if this works properly, because you don't know how deeply nested these two folders are.  Maybe you should find the top-level parent for both folders, and compare their display names?